cmd/axelard/cmd/gennetwork.go

```diff
@@ -193,9 +197,9 @@ func SetGenesisChainParamsCmd(defaultNodeHome string) *cobra.Command {
 	return cmd
 }
 
-func findEVMChain(chains []evmTypes.GenesisState_Chain, chainName string) (chain evmTypes.GenesisState_Chain, index int) {
+func findEVMChain(chains []evmTypes.GenesisState_Chain, chainName nexus.ChainName) (chain evmTypes.GenesisState_Chain, index int) {
 	for index, chain = range chains {
-		if strings.EqualFold(chainName, chain.Params.Chain) {
+		if strings.EqualFold(chainName.String(), chain.Params.Chain.String()) {
```

Comment:
Perhaps add an `Equal` method on the `ChainName` type?

Reply:
good idea